src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -383,11 +383,74 @@ public static string Combine(params string[] paths)
             return StringBuilderCache.GetStringAndRelease(finalPath);
         }
 
-        /// <summary>
-        /// Combines two paths. Does no validation of paths, only concatenates the paths
-        /// and places a directory separator between them if needed.
-        /// </summary>
-        private static string CombineNoChecks(ReadOnlySpan<char> first, ReadOnlySpan<char> second)
+        // Unlike Combine(), Join() methods do not consider rooting. They simply combine paths, ensuring that there
+        // is a directory separator between them.
+
+        public static string Join(ReadOnlySpan<char> path1, ReadOnlySpan<char> path2)
+        {
+            if (path1.Length == 0)
+                return path2.Length == 0 ? string.Empty : new string(path2);
+
+            return JoinInternal(path1, path2);
+        }
+
+        public static string Join(ReadOnlySpan<char> path1, ReadOnlySpan<char> path2, ReadOnlySpan<char> path3)
+        {
+            if (path1.Length == 0)
+                return Join(path2, path3);
+
+            if (path2.Length == 0)
+                return Join(path1, path3);
+
+            if (path3.Length == 0)
+                return Join(path1, path2);
+
+            return JoinInternal(path1, path2, path3);
+        }
+
+        public static bool TryJoin(Span<char> destination, out int charsWritten, ReadOnlySpan<char> path1, ReadOnlySpan<char> path2)
+        {
+            charsWritten = 0;
+            if (path1.Length == 0)
+            {
+                if (destination.Length < path2.Length)
+                {
+                    return false;
+                }
+
+                path2.CopyTo(destination);
+                charsWritten = path2.Length;
+                return true;
+            }
+
+            if (path2.Length == 0)
```

Comment:
Could you avoid this duplication by acting on temp, which is whichever of path1 and path2 is not empty?